Packs/TrendMicroApex/Integrations/TrendMicroApex/TrendMicroApex.py

```diff
@@ -115,47 +182,171 @@ def prodagent_restore(self, multi_match=False, entity_id="", ip_add="", mac_add=
+        return filter_entry
+
+    def create_payload_filter(self, endpoint_name, endpoint_type, ip_address, operating_system):
+        payload_filter = []
+        if endpoint_name:
+            payload_filter.append(self.create_filter_entry(entry_value=endpoint_name, entry_type="endpoint_name"))
+        if endpoint_type:
+            payload_filter.append(self.create_filter_entry(entry_value=endpoint_type, entry_type="endpoint_type"))
+        if ip_address:
+            ip_range = argToList(ip_address)
+            payload_filter.append(self.create_filter_entry(entry_value=ip_range, entry_type="endpoint_ip_address"))
+        if operating_system:
+            if operating_system == "Windows General":
+                # special case - filter by all windows security agents
+                payload_filter.append(self.create_filter_entry(entry_value=operating_system, entry_type="partial_OS"))
+            else:
+                payload_filter.append(self.create_filter_entry(entry_value=operating_system, entry_type="endpoint_OS"))
+
+        return payload_filter
+
+    def endpoint_sensors_list(self, limit=50, offset=0, filter_by_endpoint_name="", filter_by_endpoint_type="",
+                              filter_by_ip_address="", filter_by_operating_system=""):
+
+        payload_data = {
+            "pagination": {
+                "limit": int(limit),
+                "offset": int(offset)
+            }
+        }
+
+        payload_filter = self.create_payload_filter(filter_by_endpoint_name, filter_by_endpoint_type,
+                                                    filter_by_ip_address, filter_by_operating_system)
+        if payload_filter:
+            payload_data["filter"] = payload_filter
+
+        request_data = {
+            "Url": "V1/Task/ShowAgentList",
+            "TaskType": 4,  # For Endpoint Sensor, the value is always 4.
+            "Payload": payload_data
+        }
+
+        headers = {
+            'Content-Type': 'application/json;charset=utf-8',
+            'Authorization': 'Bearer ' + self.create_jwt_token(http_method='PUT', api_path=self.suffix,
+                                                               headers='', request_body=json.dumps(request_data))}
+
+        response = self._http_request("PUT", self.suffix, headers=headers, data=json.dumps(request_data))
+        return response
+
+    @staticmethod
+    def parse_cef_logs_to_dict_logs(response):
+        logs_list_in_cef_format = response.get('Data', {}).get('Logs', [])
+        parsed_logs_list = []
+        for log in logs_list_in_cef_format:
+            parsed_log = pycef.parse(log)
+            parsed_logs_list.append(parsed_log)
+
+        return parsed_logs_list
+
 
 ''' COMMANDS + REQUESTS FUNCTIONS '''
 
 
-def test_module(client: Client, *_):
+def test_module(client: Client):
     """
     Performs basic get request to get item samples
     """
-    client.usdo_list()
-    return 'ok', None, None
+    client.udso_list()
+    return 'ok'
```

Comment:
Make sure all the errors are presented correctly.

Reply:
@reutshal any update here?
Let's add fetch params here as well